new-lamassu-admin/src/pages/AddMachine/AddMachine.js

```diff
@@ -155,45 +184,51 @@ const renderStepper = (step, it, idx, classes) => {
   )
 }
 
-const AddMachine = memo(({ close }) => {
+const AddMachine = memo(({ close, onPaired }) => {
   const classes = useStyles()
-  const [qrCode, setQrCode] = useState(null)
-  const [step, setStep] = useState(0)
+  const { data } = useQuery(GET_MACHINES)
+  const [payload, setPayload] = useSetState({ qrcode: '', name: '', count: 0 })
```

Comment:
Instead of this every one gets its own state and pass down as a prop: qrCode, setQrCode, name, setName, count.